src/codegen/abi.js

```diff
@@ -594,6 +585,20 @@ module.exports = class AbiCodeGenerator {
     return [...types, klass]
   }
 
+  _getTupleParamType(inputOrOutput, index, tupleResultParentType) {
+    const type = inputOrOutput.get('type')
+    return util.isTupleType(type)
+      ? this._tupleTypeName(inputOrOutput, index, tupleResultParentType, this.abi.name)
+      : util.isTupleArrayType(type)
+      ? `Array<${this._tupleTypeName(
+          inputOrOutput,
+          index,
+          tupleResultParentType,
+          this.abi.name,
+        )}>`
+      : typesCodegen.ascTypeForEthereum(type)
```

Comment:
Since this can be used for input and result types, let's rename the parent type argument from `tupleResultParentType` to `tupleParentType`.